### PR TITLE
fix(price-feed): fix age pruning, toBN conversions and param check

### LINKED ISSUE #1955: balancer price feed has various issues
**Description**
- prices need to return bns
- need to be able to disable lookback or set to 0 without erroring
- issue when pruning by timestamp

**Reproduction**
Start monitor bot and add balancer price feed to env

**Expected Behavior**

**Screenshots**

```
9500 BN { negative: 0, words: [ 54303375, 1431772 ], length: 2, red: null }
2020-09-04 18:37:09 [debug]: {
  "at": "Monitor#index",
  "message": "An error was thrown in the execution loop - retrying",
  "bot-identifier": "usdbtcoct-kovan-monitor"
}
TypeError: observedValue.sub(...).mul is not a function
    at SyntheticPegMonitor._calculateDeviationError 
```
image

**Version and OS Info**

 - OS: [e.g. Ubuntu]
 - Version [either git hash of the repository or release]

**Additional Context**

### PR DESCRIPTION
**Pull request by daywiss**
**Motivation**

#1955 

**Summary**

Fixes some API issues and bugs with the balancer price feed that came up during wargames testing.
Additionally default balancer lookback to 0 blocks, and update block finder to add a single block even if its before age, so there is at least one. 

**Issue(s)**

Fixes #1955

### CODE REVIEW

File: packages/financial-templates-lib/src/price-feed/BalancerPriceFeed.js [hunk 1]
```diff
@@ -48,7 +50,17 @@ class BalancerPriceFeed extends PriceFeedInterface {
     this.lastUpdateTime = await this.getTime();
     // its possible provider throws error getting block history
     // we are going to just ignore these errors...
-    const blocks = await this.blockHistory.update(this.lookback, this.lastUpdateTime);
+    let blocks = [];
+    if (this.lookback <= 0) {
```
Comment (nicholaspai) [inline packages/financial-templates-lib/src/price-feed/BalancerPriceFeed.js:54]: Maybe this should just be `this.lookback === 0` ? Why would it ever be neg

**Review by nicholaspai — APPROVED** [review]
LGTM! nice fixes

File: packages/financial-templates-lib/src/price-feed/utils.js [hunk 2]
```diff
@@ -65,7 +65,7 @@ exports.BlockHistory = (getBlock, blocks = []) => {
 
   // Removes blocks from cache which are older than age
   function pruneByTimestamp(age) {
-    blocks = blocks.filter(block => block.timestamp > seconds);
```
Comment (chrismaree) [inline packages/financial-templates-lib/src/price-feed/utils.js:68]: how did this ever work?

**Review by chrismaree — APPROVED** [review]

File: packages/financial-templates-lib/src/price-feed/utils.js (hunk 2 above)
Comment (daywiss) [reply packages/financial-templates-lib/src/price-feed/utils.js:68]: Didnt test it 😅

**Pull request merged**